packages/shared/friends/sagas.ts

```diff
@@ -456,6 +459,17 @@ function* initializePrivateMessaging() {
     }
   ])
 
+  const profile: Avatar | null = yield select(getCurrentUserProfile)
+  if (profile) {
```

Comment:
Why is this necessary? The renderer already both 1) knows our profile pic and 2) can generate it from scratch

Reply:
When a user logs in, we are sending the name and image hash to Matrix, so other users that don't know us can have that information (so we don't resolve profiles as always)

Reply:
can we prevent needing this?... unity has an internal texture called `AvatarTexture(address)` that can be used at any time to get the texture based on an address. using that may remove the need of passing that information at all

Reply:
I believe we can, but what exactly do you want to prevent:

- hashing the content
- send this information to Matrix
- send image hash
- send default profiles from kernel to the explorer

or all of them?

sending just the address will imply some handling on the Explorer side, cc @sandrade-dcl @lorux0, are you OK with that?

Reply:
hmmm... I don't know any `AvatarTexture(address)` in the Unity side. @menduz maybe you're meaning the `GetAvatarUrls` function in the `DCLAvatarTexture` class? https://github.com/decentraland/unity-renderer/blob/dev/unity-renderer/Assets/Scripts/MainScripts/DCL/Components/Textures/DCLAvatarTexture.cs#L90

Reply:
As I can see, it is simply doing a request using this:

```
Environment.i.platform.serviceProviders.catalyst.lambdasUrl + "/profiles?id=" + model.userId
```

Reply:
All of them are things that I'd like to prevent. The information flows should be way easier than the proposed solution.

The best scenario is adding a `/lambdas/profile-pictures/:address` url to make the whole integration completely stateless.

All this state is damaging for the maintenance of the project(s) and increment the surface area of the responsibilities of kernel and matrix, and we may want to reduce that as much as possible.

Reply:
We do need the profile picture url for being able to display it on the member list, we will need to solve it on renderer side anyways so i dont see an advantage there. On the other hand, we dont need profile pictures to be able to display chat messages, so that case is avoidable. However this change would require a refactor on the renderer side, risking the current solution though